### PR TITLE
Feature Update Managed Variants From File

### PR DESCRIPTION
**Pull request by mikaell**
This PR adds a functionality, allowing users to update existing managed variants when loading from file. This allows for changes in description, institute and maintainer. As changing other fields would result in another id -hence not being the same variant.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

1. From Managed variants page load a file2. For example:
```
chromosome;position;end;reference;alternative;category;sub_category;description
14;76548781;76548781;CTGGACCEEE;G;snv;indel;IFT43 indel
```

2. Edit the the description (the last entry)
3. Load again, the managed variant is updated.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by DN

### CODE REVIEW

**Comment by dnil** [comment]
Tests ok!

Load and edit:
Screenshot 2022-10-06 at 11 11 14

Load again:
Screenshot 2022-10-06 at 11 11 36

File: scout/adapter/mongo/managed_variant.py [hunk 1]
```diff
@@ -231,3 +237,22 @@ def delete_managed_variant(self, managed_variant_id):
             )
 
         return result
+
+
+def _update_existing(managed_variant_a, managed_variant_b):
```
[the remark this answers is not in the document]
Comment (mikaell) [reply scout/adapter/mongo/managed_variant.py]: Done!

File: scout/adapter/mongo/managed_variant.py [hunk 2]
```diff
@@ -43,25 +43,31 @@ def upsert_managed_variant(self, managed_variant_obj, original_obj_id=None):
...
+        if collision:
+            # edit from gui, may update key contruction values
+            if original_obj_id:
+                result = self.managed_variant_collection.find_one_and_update(
+                    {"_id": ObjectId(original_obj_id)},
+                    {"$set": managed_variant_obj},
+                )
+                return True
+
+            # edit from file, write an update if key construction values are unchanged
```
[the remark this answers is not in the document]
Comment (mikaell) [reply scout/adapter/mongo/managed_variant.py]: Moved!

File: scout/adapter/mongo/managed_variant.py [hunk 3]
```diff
@@ -231,3 +233,21 @@ def delete_managed_variant(self, managed_variant_id):
             )
 
         return result
+
+
+def _non_id_values_updated(managed_variant_a, managed_variant_b):
+    """Compare two managed variants. If non-id creating values are different the updated
+    variant is written to database.
```
Comment (dnil) [inline scout/adapter/mongo/managed_variant.py:240]:
```suggestion
    variant is returned, checked ok to be written to database.
```

File: scout/build/managed_variant.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import logging
-
+from scout.exceptions import IntegrityError
```
Comment (dnil) [inline scout/build/managed_variant.py:2]:
```suggestion
```

**Review by dnil — APPROVED** [review]
Just a suggestion for that comment, and also an extra import? Thanks!

**Pull request merged**